src/main/java/io/cryostat/core/agent/ProbeTemplate.java

```diff
@@ -92,11 +92,18 @@ public ProbeTemplate() {
     }
 
     public void deserialize(InputStream xmlStream) throws IOException, SAXException {
-        BufferedInputStream stream = new BufferedInputStream(xmlStream);
-        xmlStream.mark(1); // arbitrary readLimit > 0
+        BufferedInputStream stream =
+                new BufferedInputStream(xmlStream) {
+                    @Override
+                    public void close() throws IOException {
```

Comment:
How is this stream properly closed in this case? And is the underlying `xmlStream` also closed somewhere else?

Reply:
I missed closing it, oops.

I've split this off into a new AgentXMLStream class that overrides the original close (since the XML validator closes the stream when it completes we need to override this otherwise it gets closed before we can reset) and has it's own close that we can call later when we actually want to close it.